In [19]:
#importando bibliotecas
import pandas as pd
import numpy as np
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity= "all"

In [20]:
df = pd.read_csv("insurance.csv")

In [21]:
#verificando as primeiras e as últimas linhas
df.head()
df.tail()
#verificando se há valores missing
df.info()

age     sex     bmi  children smoker     region      charges
0   19  female  27.900         0    yes  southwest  16884.92400
1   18    male  33.770         1     no  southeast   1725.55230
2   28    male  33.000         3     no  southeast   4449.46200
3   33    male  22.705         0     no  northwest  21984.47061
4   32    male  28.880         0     no  northwest   3866.85520

age     sex    bmi  children smoker     region     charges
1333   50    male  30.97         3     no  northwest  10600.5483
1334   18  female  31.92         0     no  northeast   2205.9808
1335   18  female  36.85         0     no  southeast   1629.8335
1336   21  female  25.80         0     no  southwest   2007.9450
1337   61  female  29.07         0    yes  northwest  29141.3603

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


Não há valores missing no dataset. <br>
O dataset está consumindo 73.3kb de memória

In [22]:
#Estatísiticas descritivas do dataset
df.describe()

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

BMI possui uma mediana de 30.40 e uma média de 30.66. <br>
A média de BMI é maior que a mediana, isso indica que a distribuição está assimetricamente inclinada à direita, ou seja, é uma distribuição com uma cauda mais longa no lado dos valores maiores.  <br>
No caso da coluna BMI <br>
A média do "bmi" é 30.663397.<br>
A mediana do "bmi" é 30.400000. <br>
Essa diferença sugere que há alguns valores de "bmi" significativamente altos que estão puxando a média para cima. Em outras palavras, a maioria dos valores de "bmi" está concentrada em torno da mediana, mas há alguns valores extremos (outliers) que são bem maiores e elevam a média.

In [23]:
#Proporção de fumantes
df["smoker"].value_counts(normalize = True)

smoker
no     0.795217
yes    0.204783
Name: proportion, dtype: float64

79.5% são fumantes <br>
20.4% são não fumantes

In [24]:
df["smoker"].value_counts()
df.groupby("smoker")["charges"].mean()

smoker
no     1064
yes     274
Name: count, dtype: int64

smoker
no      8434.268298
yes    32050.231832
Name: charges, dtype: float64

É possível observar que há uma difrença significativa de <i>charges</i> para fumantes, isso se deve ao maior risco de vida e doenças.

In [25]:
#Média de Bmi para fumantes
df.groupby("smoker")["bmi"].mean()

smoker
no     30.651795
yes    30.708449
Name: bmi, dtype: float64

A média de BMI para Fumantes e Não Fumantes é bem parecida.

In [26]:
df["temFilho"] = np.where(
df["children"] == 0,
0,
1)
df.head()
df.groupby("temFilho")["charges"].mean()

age     sex     bmi  children smoker     region      charges  temFilho
0   19  female  27.900         0    yes  southwest  16884.92400         0
1   18    male  33.770         1     no  southeast   1725.55230         1
2   28    male  33.000         3     no  southeast   4449.46200         1
3   33    male  22.705         0     no  northwest  21984.47061         0
4   32    male  28.880         0     no  northwest   3866.85520         0

temFilho
0    12365.975602
1    13949.941093
Name: charges, dtype: float64

Há uma diferença de 1583 cobranças a mais para quem tem 1 ou mais filhos.

In [27]:
#usando np.select eu consegui criar uma tabela que se baseasse em 3 condições para o bmi
conditions = [
    (df["sex"] == "male") & (df["bmi"]<20),
    (df["sex"] == "male") & (df["bmi"]>=20) & (df["bmi"]<= 24.9),
    (df["sex"] == "male") & (df["bmi"]>24.9)
]
choices = ["Abaixo do peso", "Normal", "Acima do peso"]
df["bmiMale"] = np.select(conditions, choices, default = '')

conditions = [
    (df["sex"] == "female") & (df["bmi"]<18.5),
    (df["sex"] == "female") & (df["bmi"]>=18.5) & (df["bmi"]<= 25),
    (df["sex"] == "female") & (df["bmi"]>25)
]
choices = ["Abaixo do peso", "Normal", "Acima do peso"]
df["bmiFemale"] = np.select(conditions, choices, default = '')
df.head(10)





age     sex     bmi  children smoker     region      charges  temFilho  \
0   19  female  27.900         0    yes  southwest  16884.92400         0   
1   18    male  33.770         1     no  southeast   1725.55230         1   
2   28    male  33.000         3     no  southeast   4449.46200         1   
3   33    male  22.705         0     no  northwest  21984.47061         0   
4   32    male  28.880         0     no  northwest   3866.85520         0   
5   31  female  25.740         0     no  southeast   3756.62160         0   
6   46  female  33.440         1     no  southeast   8240.58960         1   
7   37  female  27.740         3     no  northwest   7281.50560         1   
8   37    male  29.830         2     no  northeast   6406.41070         1   
9   60  female  25.840         0     no  northwest  28923.13692         0   

         bmiMale      bmiFemale  
0                 Acima do peso  
1  Acima do peso                 
2  Acima do peso                 
3         Normal                 
4  Acima do peso                 
5                 Acima do peso  
6                 Acima do peso  
7                 Acima do peso  
8  Acima do peso                 
9                 Acima do peso

In [28]:
df[["bmi","bmiMale", "charges", "smoker"]]

bmi        bmiMale      charges smoker
0     27.900                 16884.92400    yes
1     33.770  Acima do peso   1725.55230     no
2     33.000  Acima do peso   4449.46200     no
3     22.705         Normal  21984.47061     no
4     28.880  Acima do peso   3866.85520     no
...      ...            ...          ...    ...
1333  30.970  Acima do peso  10600.54830     no
1334  31.920                  2205.98080     no
1335  36.850                  1629.83350     no
1336  25.800                  2007.94500     no
1337  29.070                 29141.36030    yes

[1338 rows x 4 columns]

In [52]:

indicador = df[(df["sex"] == "male") & (df["smoker"] == "yes") & (df["bmiMale"] =="Acima do peso")].shape[0]
print(f"Quantidade de homens fumantes com Bmi acima do peso: {indicador}")

Quantidade de homens fumantes com Bmi acima do peso: 136
